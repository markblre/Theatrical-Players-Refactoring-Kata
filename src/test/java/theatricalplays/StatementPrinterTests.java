package theatricalplays;

import org.junit.jupiter.api.Test;

import java.util.List;

import static org.approvaltests.Approvals.verify;
import static org.approvaltests.Approvals.verifyHtml;

public class StatementPrinterTests {

  @Test
  void testTXTInvoiceWithoutDiscount() {
    Customer BigCo = new Customer("BigCo");

    Play Hamlet = new Tragedy("Hamlet");
    Play AsYouLikeIt = new Comedy("As You Like It");
    Play Othello = new Tragedy("Othello");

    Invoice invoice = new Invoice(BigCo, List.of(
            new Performance(Hamlet, 55),
            new Performance(AsYouLikeIt, 35),
            new Performance(Othello, 40)));

    StatementPrinter statementPrinter = new StatementPrinter();
    var result = statementPrinter.printTXT(invoice.getData());

    verify(result);
  }

  @Test
  void testTXTInvoiceWithDiscount() {
    Customer BigCo = new Customer("BigCo");

    BigCo.credits = 130;

    Play Hamlet = new Tragedy("Hamlet");
    Play AsYouLikeIt = new Comedy("As You Like It");
    Play Othello = new Tragedy("Othello");

    Invoice invoice = new Invoice(BigCo, List.of(
      new Performance(Hamlet, 55),
      new Performance(AsYouLikeIt, 35),
      new Performance(Othello, 40)));

    StatementPrinter statementPrinter = new StatementPrinter();
    var result = statementPrinter.printTXT(invoice.getData());

    verify(result);
  }

  @Test
  void testHTMLInvoiceWithoutDiscount() {
    Customer BigCo = new Customer("BigCo");

    Play Hamlet = new Tragedy("Hamlet");
    Play AsYouLikeIt = new Comedy("As You Like It");
    Play Othello = new Tragedy("Othello");

    Invoice invoice = new Invoice(BigCo, List.of(
      new Performance(Hamlet, 55),
      new Performance(AsYouLikeIt, 35),
      new Performance(Othello, 40)));

    StatementPrinter statementPrinter = new StatementPrinter();
    var result = statementPrinter.printHTML(invoice.getData());

    verifyHtml(result);
  }

  @Test
  void testHTMLInvoiceWithDiscount() {
    Customer BigCo = new Customer("BigCo");

    BigCo.credits = 130;

    Play Hamlet = new Tragedy("Hamlet");
    Play AsYouLikeIt = new Comedy("As You Like It");
    Play Othello = new Tragedy("Othello");

    Invoice invoice = new Invoice(BigCo, List.of(
      new Performance(Hamlet, 55),
      new Performance(AsYouLikeIt, 35),
      new Performance(Othello, 40)));

    StatementPrinter statementPrinter = new StatementPrinter();
    var result = statementPrinter.printHTML(invoice.getData());

    verifyHtml(result);
  }
}
