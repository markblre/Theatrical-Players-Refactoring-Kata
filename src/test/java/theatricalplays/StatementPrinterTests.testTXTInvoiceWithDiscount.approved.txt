Statement for BigCo
  Hamlet: $650.00 (55 seats)
  As You Like It: $580.00 (35 seats)
  Othello: $500.00 (40 seats)
Subtotal: $1,730.00
Previous credit balance: 130
You earned 47 credits
Discount: $15.00 (-150 credits)
New credit balance: 27
Total: $1,715.00
